build/templates/toolTemplates.csproj

```diff
@@ -0,0 +1,14 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```

Comment:
should we re-use the template that is already there? Or is the problem the TFM specified there?

Reply:
Reuse template, and the older TFM will pass via command line param. 

As for the name, although it is template as template engine, there is no better name than just "template"